src/command.rs

```diff
@@ -279,15 +284,19 @@ pub async fn login(mut credentials: Credentials, registry: RegistryUri) -> eyre:
         raw.trim().into()
     };
 
-    credentials.registry_tokens.insert(registry.clone(), token);
+    credentials.registry_tokens.insert(uri.clone(), token);
 
-    let artifactory = Artifactory::new(registry, &credentials)?;
+    match uri.kind() {
+        RegistryType::Artifactory => {
```

Comment:
Why do we need this?

Reply:
We don't strictly need it, but in the future there will be other kinds of registry. We shouldn't assume it's always Artifactory.

Reply:
> but in the future there will be other kinds of registry
No 😅 
There will be: now artifactory, soon the buffrs registry (which entirely replaces artifactory

Reply:
Ok, this changes things! We need a different kind of refactor then. Will close this out.